src/main/java/hudson/tasks/test/AggregatedTestResultPublisher.java

```diff
@@ -144,43 +145,41 @@ public static final class TestResultAction extends AbstractTestResultAction {
 
         private transient List<AbstractProject> noFingerprints;
 
-        @SuppressWarnings("deprecation") // calls getProject in constructor, so needs owner immediately
+        /**
+         * @deprecated use {@link #TestResultAction(String, boolean)}
+         */
+        @Deprecated(forRemoval = true)
         public TestResultAction(String jobs, boolean includeFailedBuilds, AbstractBuild<?, ?> owner) {
-            super(owner);
-            this.includeFailedBuilds = includeFailedBuilds;
+            this(jobs, includeFailedBuilds);
+        }
 
-            if (jobs == null) {
-                // resolve null as the transitive downstream jobs
-                StringBuilder buf = new StringBuilder();
-                for (AbstractProject p : getProject().getTransitiveDownstreamProjects()) {
-                    if (buf.length() > 0) {
-                        buf.append(',');
-                    }
-                    buf.append(p.getFullName());
-                }
-                jobs = buf.toString();
-            }
+        public TestResultAction(String jobs, boolean includeFailedBuilds) {
+            this.includeFailedBuilds = includeFailedBuilds;
             this.jobs = jobs;
         }
 
         /**
          * Gets the jobs to be monitored.
          */
         public Collection<AbstractProject> getJobs() {
+            if (jobs == null) {
+                Set<AbstractProject> projects = getProject().getTransitiveDownstreamProjects();
+                return projects.stream()
+                        .filter(p -> p.hasPermission(Item.READ))
+                        .collect(Collectors.toSet());
```

Comment:
need to keep the spotless overlord hapy
```suggestion
                return projects.stream().filter(p -> p.hasPermission(Item.READ)).collect(Collectors.toSet());
```